src/internal_macros.rs

```diff
@@ -18,18 +18,21 @@
 
 macro_rules! impl_consensus_encoding {
     ($thing:ident, $($field:ident),+) => (
-        impl<S: ::consensus::encode::Encoder> ::consensus::encode::Encodable<S> for $thing {
+        impl ::private::Sealed for $thing { }
+
+        impl<S: ::consensus::WriteExt> ::consensus::Encodable<S> for $thing {
             #[inline]
-            fn consensus_encode(&self, s: &mut S) -> Result<(), ::consensus::encode::Error> {
-                $( self.$field.consensus_encode(s)?; )+
-                Ok(())
+            fn consensus_encode(&self, s: &mut S) -> Result<usize, ::consensus::encode::Error> {
+                let mut len = 0;
+                $(len += self.$field.consensus_encode(s)?;)+
+                Ok(len)
             }
         }
 
-        impl<D: ::consensus::encode::Decoder> ::consensus::encode::Decodable<D> for $thing {
+        impl<D: ::consensus::ReadExt> ::consensus::Decodable<D> for $thing {
             #[inline]
             fn consensus_decode(d: &mut D) -> Result<$thing, ::consensus::encode::Error> {
-                use consensus::encode::Decodable;
+                use consensus::Decodable;
```

Comment:
Given that all type references in the macros have the `::` prefix, this one should have it too I guess.